test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/NormalizedModelElementsCacheTests.cs

```diff
@@ -157,10 +157,115 @@ public void FindTerms_ReturnsNullIfThereAreNoMatches()
             model.AddTerm("NS.Models", "SomeTerm", EdmPrimitiveTypeKind.String);
             model.AddTerm("NS.Models", "OtherTerm", EdmPrimitiveTypeKind.String);
 
-            var cache = new NormalizedSchemaElementsCache(model);
+            var cache = new NormalizedModelElementsCache(model);
 
             var matches = cache.FindTerms("ns.models.some");
             Assert.Null(matches);
         }
+
+        [Theory]
+        [InlineData("People")]
+        [InlineData("people")]
+        public void FindNavigationSources_ReturnsNavigationSourcesThatMatcheName(string name)
+        {
+            var model = new EdmModel();
+            var person = model.AddEntityType("NS", "People");
+            person.AddKeys(person.AddStructuralProperty("Id", EdmPrimitiveTypeKind.Int32));
+
+            var container = model.AddEntityContainer("NS", "Container");
+            var entitySet1 = container.AddEntitySet("People", person);
+            var entitySet2 = container.AddSingleton("peoPle", person);
+            container.AddEntitySet("Persons", person);

```

Comment:
this line,,i don't think it is necessary here.

Reply:
I added this line to ensure that the method returns only the items that match the keyword.

To explain the rationale, let's say I had not added "Persons" to the container and only had "People" and "peoPle". Now let's say that in the implementation of `cache.FindNavigationSources` I had a bug that caused every navigation source in the container to be returned, as opposed to only returning those that match. Such a bug would not be caught by this test, the test would be pass because everything in the container happens to match. So to be able to catch such bugs, I added something that does not match to the container to verify that such items are not included in the result.